### ISSUE TITLE
Null Parsing Setting?

### ISSUE THREAD

**Issue by geoffreywiseman** [uniVocity/univocity-parsers#355]
Hey,

I'm using Univocity CSV to parse some CSVs that include nulls, where the nulls are explicit:

Imagine:
```
row_name, index, value
"first row", 1, null
"second row", 2, 5
"third row", 3, null
```

If I parse these into records and say `record.getInt("value")`, I get:

`java.lang.NumberFormatException: For input string: "null"`

Essentially, it seems like the Record has treated `null` as a String instead of parsing it as a null. I've been poking around in the settings, but I don't see anything that seems to address this. There is a `nullValue`, but that seems to be about changing the value the record would use when it detects a null, whereas what I want is control over what the Record parsing considers a null.

Is there a setting for this that I'm missing?

**Comment by jbax** [comment]
Try getRecordMetadata.setTypeOfColumns(int.class, "col1", "col6", "colN");

Cheers

**Comment by geoffreywiseman** [comment]
This is a test case demonstrating my initial problem:
```java
@Test
public void testUnivocityNull() {
	String data = "name,value\n'a',1\n'b',null\n";
	CsvParserSettings settings = new CsvParserSettings();
	settings.getFormat().setQuote( '\'' );
	settings.setHeaderExtractionEnabled( true );
	CsvParser parser = new CsvParser( settings );
	List<Record> records = parser.parseAllRecords(new StringReader(data));
	assertEquals(records.get(0).getString("name"), "a");
	assertEquals(records.get(0).getInt("value"), new Integer(1));
	assertEquals(records.get(1).getString("name"), "b");
	assertNull(records.get(1).getInt("value"));
}
```

If I correctly understand your suggestion, this test case shows your suggested fix:
```
	@Test
	public void testUnivocityNull() {
		String data = "name,value\n'a',1\n'b',null\n";
		CsvParserSettings settings = new CsvParserSettings();
		settings.getFormat().setQuote( '\'' );
		settings.setHeaderExtractionEnabled( true );
		CsvParser parser = new CsvParser( settings );
		parser.beginParsing( new StringReader(data) );
		parser.getRecordMetadata().setTypeOfColumns( Integer.class, "value" );
		List<Record> records = parser.parseAllRecords();
		assertEquals(records.get(0).getString("name"), "a");
		assertEquals(records.get(0).getInt("value"), new Integer(1));
		assertEquals(records.get(1).getString("name"), "b");
		assertNull(records.get(1).getInt("value"));
	}
```

This test case fails the same way.

**Comment by jbax** [comment]
Now I realize what is the problem. Your input has a field whose value is the text `null`. When parsing that will be treated as a regular string, i.e. the parser will store the `"null"` String instead of the `null` value.

You need to instruct the parser to convert that string to the `null` value. This should do the trick:

`parser.getRecordMetadata().convertFields(Conversions.toNull("null")).set("value");`

**Comment by geoffreywiseman** [comment]
Yeah, adding conversions helps as long as I'm using `getValue()` or `getInt()`. It fails if I use `getString()`; from what I can see from looking around in the codebase, it looks like `getString()` is more of a raw value retrieval, without conversions applied?

Using the test from above:

```java
@Test
public void testUnivocityNull() {
	CsvParserSettings settings = new CsvParserSettings();
	settings.getFormat().setQuote( '\'' );
	settings.setHeaderExtractionEnabled( true );

	String data = "name,value\n'a',1\n'b',null\n";
	CsvParser parser = new CsvParser( settings );
	parser.beginParsing( new StringReader( data ) );
	parser.getRecordMetadata().convertFields( Conversions.toNull("null") ).set( "value" );
	List<Record> records = parser.parseAllRecords();

	assertEquals( records.get( 0 ).getString( "name" ), "a" );
	assertEquals( records.get( 0 ).getInt( "value" ), new Integer( 1 ) );
	assertEquals( records.get( 1 ).getString( "name" ), "b" );
	assertNull( records.get( 1 ).getInt( "value" ) );
	assertNull( records.get( 1 ).getValue( "value", String.class ) );
	assertNull( records.get( 1 ).getString( "value" ) );
}
```

In that test first two `assertNull` calls succeed, but the last one fails, because `getString()` still returns `"null"` instead of `null`.

**Comment by jbax** [comment]
Thanks for providing a test case. I've fixed the problem and released a 2.8.4-SNAPSHOT build that includes the adjustment.

Best regards
